### PR TITLE
set keys in gb tree to iterator value

### PR DESCRIPTION
**Pull request by pringlewood**
Do not use random string for key value, does not sort properly.

### CODE REVIEW

**Comment by louisvisser** [comment]
Think this repo doesn't have workflows setup; `mix test` fails:
```
Finished in 0.6 seconds (0.6s async, 0.00s sync)
16 tests, 10 failures
```

**Comment by Tom-Wyszkowski** [comment]
Breaks liveview for me 🙅‍♂️

File: lib/searchable_select.ex [hunk 1]
```diff
@@ -197,10 +197,10 @@ defmodule SearchableSelect do
...
 
   def prep_options(%{assigns: assigns} = socket, %{options: options}) do
-    gb_options =
-      Enum.reduce(options, :gb_trees.empty(), fn option, acc ->
-        normalised_label = assigns.label_callback.(option) |> normalise_string()
-        :gb_trees.insert(normalised_label, option, acc)
+    [_ , gb_options] =
+      Enum.reduce(options, [0, :gb_trees.empty()], fn option, [count, tree] ->
+        searchable_string = assigns.label_callback.(option) |> normalise_string()
+        [count + 1, :gb_trees.insert(count, {option, searchable_string}, tree)]
```
Comment (pringlewood) [inline lib/searchable_select.ex:203]: to do least about of damage to test:

`        [count + 1, :gb_trees.insert("#{count}-#{searchable_string}", option, tree)]`

**Comment by louisvisser** [comment]
@pringlewood Cool if I close this?

**Pull request closed without merging**